content/blueprints/gcp.md

```diff
@@ -0,0 +1,1277 @@
+
+    *default:* {}
+  * `use_external_resource`
+    Indicate whether the resource exists and use existing (true) or if Cloudify should create new resource (false).
+
+    *type:* boolean
+    *default:* False
+  * `name`
+    Optional instance name. By default it will be instance group id.
+
+    *type:* string
+    *default:*
+  * `named_ports`
+    A list of named ports defined for this instance group, the expected format is: [{name: 'name', port: 1234}, ... ].
+
+    *default:* []
+
+
+
+
+
+## cloudify.gcp.nodes.KeyPair
+**Derived From:** [cloudify.nodes.Root]({{< relref "blueprints/built-in-types.md" >}})
+
+An SSH key-pair which will be uploaded to any Instances connected to it via `cloudify.gcp.relationships.instance_connected_to_keypair`.
+
+Unlike other cloud providers, users are dynamically created on Instances based on the username specified by the uploaded SSH key, so the public key text must include a username in the comment section (keys generated using `ssh-keygen` have this by default).
+
+
+
+**Properties:**
+
+
+  * `private_key_path`
+    The path where the key should be saved on the machine. If this is a bootstrap process, this refers to the local computer. If this will run on the manager, this will be saved on the manager.
+
+    *type:* string
+    *default:*
+  * `public_key_path`
+    The path to read from existing public key.
+
+    *type:* string
+    *default:*
+  * `user`
+    The user account for this key. A corresponding user account will be created by GCP when the key is added to the Instance. This must be supplied for a non-external resource key. See https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys
+
+    *type:* string
+    *default:*
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+
+    *default:* {}
+  * `use_external_resource`
+    Indicate whether the resource exists or if Cloudify should create the resource.
+
+    *type:* boolean
+    *default:* False
+
+
+
+
+
+## cloudify.gcp.nodes.Network
+**Derived From:** [cloudify.nodes.Network]({{< relref "blueprints/built-in-types.md" >}})
+
+A GCP Network. This supports either auto-assigned or manual subnets. Legacy networks are not supported. See the GCP Manager and Networks section below if you plan to run a cloudify manager on GCP.
+
+
+
+**Properties:**
+
+
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+
+    *default:* {}
+  * `auto_subnets`
+    Whether to use the GCP "autoCreateSubnetworks" feature (see https://cloud.google.com/compute/docs/subnetworks#networks_and_subnetworks)
+
+    *default:* True
+  * `additional_settings`
+    Additional setting for network
+
+    *default:* {}
+  * `name`
+    Optional Network name. The instance ID will be used by default.
+
+    *default:*
+  * `use_external_resource`
+    Indicate whether the resource exists or if Cloudify should create the resource.
```

Comment:
Indicate**s** whether the resource exists **and should be used (true),** or if Cloudify should create the resource **(false).**